codegenerator/cli/templates/dynamic/codegen/src/db/Entities.res.hbs

```diff
@@ -11,9 +11,9 @@ module type Entity = {
   let table: Table.table
   let entityHistory: EntityHistory.t<t>
 }
-module type InternalEntity = Entity with type t = internalEntity
-external entityModToInternal: module(Entity with type t = 'a) => module(InternalEntity) = "%identity"
-external entityModsToInternal: array<module(Entity)> => array<module(InternalEntity)> = "%identity"
+external entityModToInternal: module(Entity with type t = 'a) => Internal.entityConfig = "%identity"
+external entityModsToInternal: array<module(Entity)> => array<Internal.entityConfig> = "%identity"
+external entitiesToInternal: array<'a> => array<Internal.entity> = "%identity"
```

Comment:
I'm a little worried about type casting modules to records. Not sure if there's any possibility of unexpected compiled differences between the two.

Do we need the module at all if we are changing to records or is it just about keeping naming conventions with uppercase module etc?

Reply:
It's just to keep the change smaller. Ultimately, we should also convert the `module Entity` into a record.